README.md

```diff
@@ -98,23 +100,47 @@ This will copy the GitHub repository contents into a new folder in the current d
 cd oot
 ```
 
-#### 3. Prepare a base ROM
+#### 3. Install python dependencies
+
+The build process has a few python packages required that are located in `requirements.txt`.
+
+It is recommend to setup a virtual environment for python to localize all dependencies. To create a virtual environment:
+
+```bash
+python3 -m venv .venv
+```
+
+To start using the virtual environment on your current terminal run:
+
+```bash
+. .venv/bin/activate
+```
+
+Take in mind for each new terminal you'll need to **active** the Python virtual environment again. There's no need to create the virtual environment again.
+
+Now you can install the Python dependencies, to do so run:
+
+```bash
+python3 -m pip install -U -r requirements.txt
```

Comment:
Why not `pip install -r requirements.txt`? That's what basically every other Python project uses so it's ingrained in my muscle memory

Reply:
Using `pip` instead of `python3 -m pip` can be problematic if you have more than one version of Python installed, which can happen if either you install it manually or your distro decides to push a newer version.
It is better practice to do this, but if anybody wants to just type `pip` then it should work in most cases.

As a safe-ish option you can slap `alias pip="python3 -m pip"` on your `~/.bashrc` file and forget about this issue.

Reply:
Yeah I also wanted to suggest `pip` over `python3 -m pip` but couldn't find solid reference on it. Everything I found even used `python3 -m pip`, including the "official" pip tutorial

Reply:
`pip` here refers to the command in the virtualenv, not the distro one

Reply:
> `pip` here refers to the command in the virtualenv, not the distro one

Yeah I haven't found reference on that

I do use it typically but since I can't find it documented I wouldn't advocate document using it

Reply:
The official docs at https://docs.python.org/3/library/venv.html say

> Changed in version 3.4: Installs pip by default, added the `--without-pip` and `--copies` options

and later it suggests using `pip install -r requirements.txt` (though oddly, only in a warning)

Reply:
bump

Reply:
https://poonam-jangid01.medium.com/to-clarify-when-you-use-it-you-are-invoking-the-pip-module-as-a-script-within-the-python-13170aa3f59f seems to be a good write up about the differences between them both. Tbh I don't really have a preference either way and not sure it really matters which we document (though `python3 -m pip` seems the "safer" option.)

Reply:
That doesn't seem to be a particularly high-quality source. What does "multiple virtual environments" even mean? If you could have multiple virtual environments, wouldn't `python` be just as ambiguous as `pip`?

Reply:
> What does "multiple virtual environments" even mean?

You can create as many virtual environments as you would like (different python versions, different packages based on projects), so it is very possible people have multiple virtual environments "installed" though only 1 can be active.
As stated in the article `python` will always point to the current activated python interpreter, so you can be assured which python is being used. On the other hand if you do use `pip` it will use the first pip it finds in the system path. When activating a virtual environment it puts that venv's bin folder first in the path so if it has pip, it would be fine since that pip should be found first. However if it doesn't, it would mess with whatever pip it does find on your system path which might cause issues.
New venv's seem to have pip installed by default now, so again either should work for our use case.